### ISSUE TITLE
Idea: Check minimum requested dependency versions instead of lockfile / highest possible

### ISSUE THREAD

**Issue by EdJoPaTo** [taiki-e/cargo-hack#254]
In the `Cargo.toml` the minimum required version can be specified. Cargo uses the highest possible one with that requirement. Especially for libraries it seems a good idea to have the correct minimal dependency version correctly. Rust Edition 2024 will also help with rust-version compatibility so "highest possible"  might be lower than expected.
Having a correct minimum will result in cargo being able to notice this conflict on dependency validation rather than on build.

Basically replacing `dependency-name = "0.1.2"` with `"=0.1.2"` and run the tests.

This could work similarly to features like `--no/remove-dev-deps`. Adapt the `Cargo.toml` and revert when done.

**Comment by EdJoPaTo** [comment]
… https://github.com/taiki-e/cargo-minimal-versions :grimacing:
